scalatest/src/main/scala/org/scalatest/tools/DashboardReporter.scala

```diff
@@ -513,8 +513,8 @@ private[scalatest] class DashboardReporter(directory: String,
   //
   def writeRunFile(event: Event, thisRunFile: File): Unit = {
     index = 0
-    var suiteRecord: SuiteRecord = null
-    val stack = new Stack[SuiteRecord]
+    var suiteRecord: SuiteRecord | Null = null
+    val stack = new Stack[SuiteRecord | Null]
```

Comment:
**Nullable collection**
Variables in collection type should be nullable. We use `new Collection[T | Null]` instead of `new Collection[T]`.